### ISSUE TITLE
'NoneType' object is not subscriptable

### ISSUE THREAD

**Issue by abhijeet0007** [matlink/gplaycli#126]
gplaycli -pd com.ubercab
Traceback (most recent call last):
  File "/usr/local/bin/gplaycli", line 9, in <module>
    load_entry_point('GPlayCli==3.10', 'console_scripts', 'gplaycli')()
  File "/usr/local/lib/python3.5/dist-packages/GPlayCli-3.10-py3.5.egg/gplaycli/                                                                                        gplaycli.py", line 528, in main
  File "/usr/local/lib/python3.5/dist-packages/GPlayCli-3.10-py3.5.egg/gplaycli/                                                                                        gplaycli.py", line 399, in download_packages
  File "/usr/local/lib/python3.5/dist-packages/GPlayCli-3.10-py3.5.egg/gplaycli/                                                                                        gplaycli.py", line 291, in download_selection
TypeError: 'NoneType' object is not subscriptable

most of the time i got this type of error
@matlink please help me out 

Please provide those informations:

- [ Ubuntu 16.04] Operating System (ie Ubuntu 16.04)
- [ Python 3.5.2 and Python 2.7.12] Python version when running `gplaycli` (should be 2.7 for now)
- [ GPlayCli version 3.10] GPlayCli version via `gplaycli -v`
- [ git clone and setup.py and pip install gplaycli] The way you installed `gplaycli`: `git clone`, `git clone and setup.py`, `pip install gplaycli`, `apt install gplaycli`, ...
- [ the authentication you provide] The authentication method: (own) credentials or (own) token.

**Comment by abhijeet0007** [comment]
hey @matlink i am still stuck with the same problem, sometime apk get downloaded but most of the time the gplaycli show the TypeError: 'NoneType' object is not subscriptable error

**Comment by matlink** [comment]
I'm on it

**Comment by matlink** [comment]
Could you paste here the output of ```gplaycli -v -pd com.ubercab``` please?

**Comment by abhijeet0007** [comment]
root@localhost:~# gplaycli -v -pd com.ubercab
[INFO] GPlayCli version 3.10 [Python3.5.2]
[INFO] Configuration file is /root/.config/gplaycli/gplaycli.conf
[INFO] Using cached token.
[INFO] Using token to connect to API
[INFO] Token has expired or is invalid. Retrieving a new one...
[INFO] Retrieving token ...
[INFO] Token: YAXaM_0oZJufr4Ag-eGOwi4X8gvRBXVwhZCpU01VK5N0Q9CInZYcczm59hGtK1vTyWOysg.
[INFO] GSFId: 3e7dabc507f5dca4
[INFO] 1 / 1 com.ubercab
Traceback (most recent call last):
  File "/usr/local/bin/gplaycli", line 11, in <module>
    load_entry_point('GPlayCli==3.10', 'console_scripts', 'gplaycli')()
  File "/usr/local/lib/python3.5/dist-packages/gplaycli/gplaycli.py", line 528, in main
    cli.download_packages(args.packages_to_download)
  File "/usr/local/lib/python3.5/dist-packages/gplaycli/gplaycli.py", line 399, in download_packages
    self.after_download)
  File "/usr/local/lib/python3.5/dist-packages/gplaycli/gplaycli.py", line 291, in download_selection
    vc = detail['versionCode']
TypeError: 'NoneType' object is not subscriptable
![1](https://user-images.githubusercontent.com/14326082/32455324-f3256678-c347-11e7-8531-a2e6b19c093c.JPG)

**Comment by singron** [comment]
It looks like the `bulkDetails` API gives a response with an empty entry. Here is what I get:
```sh
protoc --decode=ResponseWrapper googleplay.proto < out.pb
```
```protobuf
payload {
  bulkDetailsResponse {
    entry {
    }
  }
}
5 {
  1: 6
}
9 {
  1: 18
  19 {
    1 {
      1: 1510891740324001
      2: 0x0a62668a
      3: 0xc80dca6a
    }
    2: 1
  }
}
```

**Comment by singron** [comment]
I only get this error with token=True in the config. It works with token=False (i.e. password credentials).

**Comment by matlink** [comment]
Sounds like one of my Google account has been banned, I guess.

**Comment by matlink** [comment]
@abhijeet0007 can you confirm that it works with `token=False` and using your credentials?

**Comment by abhijeet0007** [comment]
@singron thank you for giving me the solution,
@matlink yes it is working good with token= False using own credentials

**Comment by gzeuro** [comment]
@matlink With token=True and version 3.11 it works now for me ... but not always. On a server with German IP address and language de_DE in the config file it works 99% of the time, on another server with US IP address only after 3-15 retries. - Maybe all these "'NoneType' object is not subscriptable errors" depends only from the server location and the current workload of the Google API in this geographical region?

**Comment by matlink** [comment]
I've pushed the fix to the tokenexpiration branch.
Could you try again with that branch?

**Comment by matlink** [comment]
Fixed in 3.12
